### ISSUE TITLE
`py38_images` tests failing after recent merge in Main 

### ISSUE THREAD

**Issue by DhruvSondhi** [poliastro/poliastro#1142]
🐞 **Problem**

The `Main` branch & subsequently all rebased branches on the current `main` are failing the `py38_images` test. The problem seems to be that the images from `test_porkchop_plotting` in `test_porkchop.py` are different from those in the assertion ... The can be seen in the code snippet below as well as this [link](https://dev.azure.com/poliastro/c2409219-95a2-4e20-b99c-187990571f10/_apis/build/builds/885/logs/66) 

```bash
=================================== FAILURES ===================================
____________________________ test_porkchop_plotting ____________________________
Error: Image files did not match.
  RMS Value: 23.55042462462195
  Expected:  
    /tmp/tmpi6yv37y_/test_porkchop.test_porkchop_plotting/baseline.png
  Actual:    
    /tmp/tmpi6yv37y_/test_porkchop.test_porkchop_plotting/result.png
  Difference:
    /tmp/tmpi6yv37y_/test_porkchop.test_porkchop_plotting/result-failed-diff.png
  Tolerance: 
    2

```

💡 **Possible solutions**

We need to check for the reason why the `test` is failing in the first place & find the commit that was recently merged leading to failure of the required `test`. Appropriate changes need to be done if the `test` has wrong assertion, upload the correct image or revert the commit so as to restructure it with all the tests passing

**Comment by xlxs4** [comment]
May be related: https://github.com/matplotlib/pytest-mpl/issues/91

**Comment by astrojuanlu** [comment]
We need to find out whether this was caused by our commit or by a new version of a dependency.

**Comment by DhruvSondhi** [comment]
The test also fails in the local virtual environment run ... Seems to be something related to the libraries 🤔 
```bash
============================================================== FAILURES ===============================================================
_______________________________________________________ test_porkchop_plotting ________________________________________________________
Error: Image files did not match.
  RMS Value: 23.55042462462195
  Expected:  
    /tmp/tmp8c_srs2q/test_porkchop.test_porkchop_plotting/baseline.png
  Actual:    
    /tmp/tmp8c_srs2q/test_porkchop.test_porkchop_plotting/result.png
  Difference:
    /tmp/tmp8c_srs2q/test_porkchop.test_porkchop_plotting/result-failed-diff.png
  Tolerance: 
    2
```

**Comment by DhruvSondhi** [comment]
I have created the images after running the invocation commands ... The images are as follows: 
_**BASELINE**_
![baseline](https://user-images.githubusercontent.com/66117751/112932610-d35b3500-913b-11eb-8ad1-0686a76ed724.png)
_**RESULT**_
![result](https://user-images.githubusercontent.com/66117751/112932615-d6562580-913b-11eb-96ef-c6dbbaaca77e.png)
_**RESULT_FAILED_DIFF**_
![result-failed-diff](https://user-images.githubusercontent.com/66117751/112932616-d7875280-913b-11eb-8d5d-4b2b6cbc3961.png)

It seems so that the problem here is the font, as mentioned appropriately by @xlxs4, as the plots don't show any difference 🤔

**Comment by astrojuanlu** [comment]
matplotlib 3.4.0 was released four days ago https://pypi.org/project/matplotlib/3.4.0/ please check whether the correct builds were using 3.3.*

**Comment by xlxs4** [comment]
From the issue I linked (and `mpl-compare`'s docs):
> Different versions of Matplotlib and FreeType may result in slightly
different images. When testing on multiple platforms or as part of a
pipeline, it is important to ensure that the versions of these
packages match the versions used to generate the images used for
comparison. It can be useful to pin versions of Matplotlib and FreeType
so as to avoid automatic updates that fail tests.

regardless of what this turns out to be, let's avoid getting bitten by this in the future

**Comment by astrojuanlu** [comment]
Fair enough, but first we ought to understand what is the root cause. If we loosen up the tolerance, we might miss legitimate bugs.

**Comment by DhruvSondhi** [comment]
Tests are failing again after downgrading `Matplotlib` to 3.3.4 😢

**Comment by DhruvSondhi** [comment]
@astrojuanlu ... Yes the Build passes without any errors with `Matplotlib==3.3.4` ... I have pinned the library version to 3.3.4 for the time being ... I will be confirming soon what the root cause may be :) & @xlxs4 is absolutely correct that for current builds, we can pin the library version of `Matplotlib` & `FreeType`
```bash
tests/tests_plotting/test_misc.py::test_plot_inner_solar_system_static PASSED                                                   [  8%]
tests/tests_plotting/test_misc.py::test_plot_outer_solar_system_static PASSED                                                   [ 16%]
tests/tests_plotting/test_porkchop.py::test_porkchop_plotting PASSED                                                            [ 25%]
tests/tests_plotting/test_static.py::test_basic_plotting PASSED                                                                 [ 33%]
tests/tests_plotting/test_static.py::test_basic_trajectory_plotting PASSED                                                      [ 41%]
tests/tests_plotting/test_static.py::test_basic_orbit_and_trajectory_plotting PASSED                                            [ 50%]
tests/tests_plotting/test_static.py::test_trail_plotting PASSED                                                                 [ 58%]
tests/tests_plotting/test_static.py::test_plot_different_planes PASSED                                                          [ 66%]
tests/tests_plotting/test_static.py::test_body_plotting PASSED                                                                  [ 75%]
tests/tests_plotting/test_static.py::test_plot_ephem_epoch SKIPPED (need --remote-data option to run)                           [ 83%]
tests/tests_plotting/test_static.py::test_plot_ephem_no_epoch SKIPPED (need --remote-data option to run)                        [ 91%]
tests/tests_plotting/test_tisserand.py::test_tisserand_plotting PASSED                                                          [100%]
```

**Comment by astrojuanlu** [comment]
These tests are just a way to check internal consistency. I propose that we regenerate the images with latest matplotlib so we can make the tests pass without changing anything else.

To regenerate the images: https://github.com/matplotlib/pytest-mpl/#with-baseline-images

**Comment by astrojuanlu** [comment]
Closed in #1144
